pkg/plumber/plumber.go

```diff
@@ -66,23 +71,36 @@ func (b *PipelineBuilder) Create(request *PipelineOptions) (*PipelineOptions, er
 	}))
 	l.Info("about to start Jenkinx X meta pipeline")
 
-	po := create.StepCreatePipelineOptions{
-		SourceURL: sourceURL,
-		Job:       job,
-		PullRefs:  pullRefs,
-		Context:   spec.Context,
-	}
 	sa := os.Getenv("JX_SERVICE_ACCOUNT")
 	if sa == "" {
 		sa = "tekton-bot"
 	}
-	po.CommonOptions = b.commonOptions
-	po.ServiceAccount = sa
 
-	err := po.Run()
+	pipelineCreateParam := metapipeline.PipelineCreateParam{
+		PullRef:      pullRefData,
+		PipelineKind: kind,
+		Context:      spec.Context,
+		// No equivalent to https://github.com/jenkins-x/jx/blob/bb59278c2707e0e99b3c24be926745c324824388/pkg/cmd/controller/pipeline/pipelinerunner_controller.go#L236
+		//   for getting environment variables from the prow job here, so far as I can tell (abayer)
+		// Also not finding an equivalent to labels from the PipelineRunRequest
+		ServiceAccount: sa,
+		// I believe we can use an empty string default image?
+		DefaultImage: "",
+	}
+
+	metaPipelineClient, err := metapipeline.NewMetaPipelineClient()
```

Comment:
I don't know the whole context of how this is exactly called, but I assume this is called on each pipeline trigger. Ideally one wants to cache the meta pipeline client opposed to creating one on request. I am not familiar with the rest of the code, but is this Plumber also called on each request? Or is it kept around? If so, one could probably create the client as part of the plumber.

Reply:
BTW, I still believe that architecturally we would be better off not depending directly on jx and the meta pipeline client. Instead we should be using the pipelinerunner and call it from Lighthouse.

Reply:
I think this is a good point. it seems to be executed on every request. See the https://github.com/jenkins-x/lighthouse/blob/e3d11e12ff05f7d086a8e90a8050431b647ebe8e/pkg/prow/plugins/trigger/trigger.go#L231.